### ISSUE TITLE
gccgo: internal compiler error in do_get_backend (3)

### ISSUE THREAD

**Issue by dvyukov** [golang/go#12617]
gccgo crashes on the following program:
```go
var _,_=<-chan int(n
```
```
go1: internal compiler error: in do_get_backend, at go/gofrontend/expressions.cc:3469
0x5e33f4 Unsafe_type_conversion_expression::do_get_backend(Translate_context*)
	../../gcc/go/gofrontend/expressions.cc:3469
0x5fcc0a Call_expression::do_get_backend(Translate_context*)
	../../gcc/go/gofrontend/expressions.cc:9373
0x64f522 Temporary_statement::do_get_backend(Translate_context*)
	../../gcc/go/gofrontend/statements.cc:487
0x61db47 Block::get_backend(Translate_context*)
	../../gcc/go/gofrontend/gogo.cc:5867
0x64ccfc Block_statement::do_get_backend(Translate_context*)
	../../gcc/go/gofrontend/statements.cc:1836
0x61db47 Block::get_backend(Translate_context*)
	../../gcc/go/gofrontend/gogo.cc:5867
0x61dcf3 Variable::get_init_block(Gogo*, Named_object*, Bvariable*)
	../../gcc/go/gofrontend/gogo.cc:6431
0x626547 Gogo::write_globals()
	../../gcc/go/gofrontend/gogo.cc:1418
```
gofrontend is on commit 1d9d92ab09996d2f7795481d2876a21194502b89

**Comment by dvyukov** [comment]
@ianlancetaylor

**Comment by paranoiacblack** [comment]
This was fixed in #12320.
